arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/halt/ReturnSection.java

```diff
@@ -302,6 +311,28 @@ public void resolvePostTransaction(
     this.addFragment(squashParentContextReturnData);
   }
 
+  private void addDeploymentAccountFragmentIfRoot(Hub hub, MxpCall mxpCall) {
+    // in case of zero depth we don't have a ContextReEntry step so we have to add the
+    // deployment account fragment manually
+    postDeploymentAccountSnapshot = AccountSnapshot.canonical(hub, deploymentAddress);
+    postDeploymentAccountSnapshot.code(
+        new Bytecode(
+            hub.messageFrame()
+                .shadowReadMemory(
+                    Words.clampedToLong(mxpCall.offset1), Words.clampedToLong(mxpCall.size1))));
+    postDeploymentAccountSnapshot.deploymentStatus(false);
+
+    final AccountFragment deploymentAccountFragment =
+        hub.factories()
+            .accountFragment()
+            .make(
+                preDeploymentAccountSnapshot,
+                postDeploymentAccountSnapshot,
+                DomSubStampsSubFragment.standardDomSubStamps(this.hubStamp(), 0));
+
+    this.addFragment(deploymentAccountFragment);
+  }
+
```

Comment:
The purpose of this method is to add the "deployment" `AccountFragment` in case the current context is the root and we are returning (successfully) from a deployment transaction. The point is: in this scenario we don't pass through `traceContextReEntry`.

Reply:
As suggested by @letypequividelespoubelles we could alternatively carry out this step in the `traceContextExit`. So far this isn't what we do.